src/events/events.service.ts

```diff
@@ -57,20 +57,26 @@ export class EventsService {
     hasPrevious: boolean;
   }> {
     const cursorId = options.before ?? options.after;
-    const cursor = cursorId ? { id: cursorId } : undefined;
     const direction = options.before !== undefined ? -1 : 1;
     const limit =
       direction * Math.min(MAX_LIMIT, options.limit || DEFAULT_LIMIT);
     const orderBy = {
       occurred_at: Prisma.SortOrder.desc,
     };
-    const skip = cursor ? 1 : 0;
+    const skip = cursorId ? 1 : 0;
+    const fn = options.before ? 'gte' : 'lte';
     const where = {
       user_id: options.userId,
       deleted_at: null,
+      ...(cursorId
```

Comment:
This is a good change, but I think it's worth separating this additional check to make it more readable. What do you think of the following?

```ts
const where: Prisma.EventWhereInput = {
  user_id: options.userId,
  deleted_at: null,
};
if (cursorId) {
  if (options.before) {
    where.id = { gte: cursorId };
  } else {
    where.id = { lte: cursorId };
  }
}
```

Reply:
yeah, it's much easier to read.